src/rpc/mining.cpp

```diff
@@ -171,6 +183,10 @@ static UniValue generatetoaddress(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Error: Invalid address");
     }
 
+    if (g_signet_blocks) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Generating signet blocks require that you use getnewblockhex");
```

Comment:
is there really no way to make it works?

Reply:
Recently, the wallet code and the rest of the code have been separated. Unfortunately this means the mining part and the signing part have to be separated as well.

Reply:
I don't understand. Why are you talking about wallet code? `generatetoaddress` does touch only mining code.

Reply:
I think we discussed this outside of here, but basically, you don't have the private keys associated with any addresses, so you'd have to provide those every time here. I don't think we should encourage systems that behave in that manner ("bitcoin core does it so it must be safe").